### ISSUE TITLE
Error when using static return type hint

### ISSUE THREAD

**Issue by milewski** [prettier/plugin-php#1834]
https://wiki.php.net/rfc/static_return_type

```php
trait Resolvable
{

    public static function resolve(): static
    {
        return resolve(static::class);
    }

}
```

It doesn't seems to be working properly .. it throws the following error:

```
SyntaxError: Parse Error : syntax error, unexpected 'static' (T_STATIC), expecting '{' on line 104
```

**Comment by alexander-akait** [comment]
Feel free to send a fix to parser, should be not hard

**Comment by czosel** [comment]
Fixed in `v0.17.6` :tada:
